### PR TITLE
Feat: test/JBETHPaymentTerminal

### PR DESCRIPTION
**Pull request by drgorillamd**

### CODE REVIEW

**Review by dhyon — COMMENTED** [review]
<redacted>

File: test/jb_eth_payment_terminal/redeem_tokens_of.test.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    // Terminal should be out of ETH
+    expect(await ethers.provider.getBalance(jbEthPaymentTerminal.address)).to.equal(0);
+
+    // Beneficiary should have a larger balance
+    expect(await ethers.provider.getBalance(beneficiary.address)).to.equal(
+      initialBeneficiaryBalance.add(AMOUNT),
+    );
+  });
+
+  it('Should only emit event if claim amount is 0', async function () {
```
Comment (mejango) [inline test/jb_eth_payment_terminal/redeem_tokens_of.test.js:162]: confused about the description string here. What event should only be emitted is claim amount is 0?

Comment (mejango) [reply test/jb_eth_payment_terminal/redeem_tokens_of.test.js:162]: the event should still be emitted, but no amount transferred to the beneficiary. The test looks right, the string off.

File: test/jb_eth_payment_terminal/use_allowance_of.test.js [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+        projectOwner.address,
+        'Fee from @' + ethers.utils.parseBytes32String(HANDLE) + PADDING,
+      ],
+    ]);
+
+    // Process held fees
+    const tx = await jbEthPaymentTerminal.connect(projectOwner).processFees(PROJECT_ID);
+
+    expect(await tx).to.emit(jbEthPaymentTerminal, 'ProcessFees');
+    // .withArgs(
```
Comment (mejango) [inline test/jb_eth_payment_terminal/use_allowance_of.test.js:408]: why's this commented out? can it be removed or does it add context?

File: test/jb_eth_payment_terminal/distribute_payouts_of.test.js [hunk 3]
```diff
@@ -0,0 +1,963 @@
...
+      mockJbDirectory,
+      mockJbEthPaymentTerminal,
+      mockJbEthPaymentTerminalStore,
+      mockJbProjects,
+      mockJbSplitsStore,
+      timestamp,
+    }
+  }
+
+  it('Should distribute payout and emit event, without fee is fee is set at 0', async function () {
```
Comment (mejango) [inline test/jb_eth_payment_terminal/distribute_payouts_of.test.js:169]: clean string

**Review by mejango — CHANGES_REQUESTED** [review]
a few touch ups. but goddamn yall this is heroic. test cases seem strong.

File: test/jb_eth_payment_terminal/use_allowance_of.test.js (hunk 2 above)
Comment (dhyon) [reply test/jb_eth_payment_terminal/use_allowance_of.test.js:409]: Added comment for context

File: test/jb_eth_payment_terminal/distribute_payouts_of.test.js [hunk 4]
```diff
@@ -0,0 +1,1094 @@
...
+    await mockJbDirectory.mock.primaryTerminalOf
+      .withArgs(1, ETH_ADDRESS)
+      .returns(mockJbEthPaymentTerminal.address);
+
+    await mockJbEthPaymentTerminal.mock.pay
+      .withArgs(
+        1, //JBX Dao
+        projectOwner.address,
+        0,
+        false,
```
Comment (dhyon) [inline test/jb_eth_payment_terminal/distribute_payouts_of.test.js:459]: bools should have an arg comment at least

File: test/jb_eth_payment_terminal/distribute_payouts_of.test.js [hunk 5]
```diff
@@ -0,0 +1,1094 @@
...
+      .withArgs(OTHER_PROJECT_ID, ETH_ADDRESS)
+      .returns(jbEthPaymentTerminal.address);
+
+    await mockJbSplitsStore.mock.splitsOf
+      .withArgs(PROJECT_ID, timestamp, ETH_PAYOUT_INDEX)
+      .returns(splits);
+
+    await Promise.all(
+      splits.map(async (split) => {
+        console.log(
```
Comment (dhyon) [inline test/jb_eth_payment_terminal/distribute_payouts_of.test.js:802]: don't need this anymore :)

**Review by dhyon — COMMENTED** [review]
lookin goood

File: test/jb_eth_payment_terminal/distribute_payouts_of.test.js (hunk 5 above)
Comment (drgorillamd) [reply test/jb_eth_payment_terminal/distribute_payouts_of.test.js]: Haha, indeed ':)

**Review by dhyon — APPROVED** [review]
aww yeah

File: test/jb_eth_payment_terminal/add_to_balance_of.test.js [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    )
+
+    await mockJbProjects.mock.ownerOf.withArgs(PROJECT_ID).returns(projectOwner.address);
+    await mockJbProjects.mock.handleOf.withArgs(PROJECT_ID).returns(HANDLE);
+
+    await mockJbEthPaymentTerminalStore.mock.recordAddedBalanceFor
+      .withArgs(PROJECT_ID, AMOUNT)
+      .returns(fundingCycle);
+
+    await ethers.provider.send('hardhat_setBalance', [jbEthPaymentTerminal.address, AMOUNT.toHexString()]);
```
Comment (odd-amphora) [inline test/jb_eth_payment_terminal/add_to_balance_of.test.js:113]: Let's put this in our helpers file

**Review by odd-amphora — APPROVED** [review]

**Pull request merged**